text_generation/causal_lm/cpp/beam_search_causal_lm.cpp

```diff
@@ -1,232 +1,38 @@
-    }
-
-    // Get the runtime info from the tokenizer model that we read earlier
-    auto rt_info = tokenizer_model->get_rt_info();  // Get the runtime info for the model
-    int64_t SPECIAL_EOS_TOKEN;
-
-    if (rt_info.count("eos_token_id") > 0) {  // check if the runtime information has a valid EOS token ID
-        SPECIAL_EOS_TOKEN = rt_info["eos_token_id"].as<int64_t>();
-
-    } else {
-        throw std::runtime_error("EOS token ID not found in model's runtime information.");
-    }
-
-    Parameters parameters{std::move(prompts), SPECIAL_EOS_TOKEN};
-    GroupBeamSearcher group_beam_searcher{parameters};
-
-    initialize_inputs(input_ids, attention_mask, lm);
-
-    std::vector<int64_t> next_tokens;
-    std::vector<int32_t> next_beams;
-
-    for (size_t length_count = 0; length_count < parameters.max_new_tokens; ++length_count) {
-        lm.infer();
-
-        std::tie(next_tokens, next_beams) = group_beam_searcher.select_next_tokens(lm.get_tensor("logits"));
-        if (next_tokens.empty()) {
-            break;
-        }
-        size_t batch_size = next_tokens.size();
-        // Set pointers
-        lm.set_tensor("input_ids", ov::Tensor{ov::element::i64, {batch_size, 1}, next_tokens.data()});
-        lm.set_tensor("beam_idx", ov::Tensor{ov::element::i32, {batch_size}, next_beams.data()});
-        // Set auxiliary inputs
-        set_attention_mask(lm.get_tensor("attention_mask"), next_beams);
-        set_position_ids(lm.get_tensor("position_ids"), lm.get_tensor("attention_mask"));
-    }
-
-    for (const std::vector<std::vector<Beam>>& prompt_group : finalize(std::move(group_beam_searcher))) {
-        std::cout << "Prompt:\n";
-        for (const std::vector<Beam> group : prompt_group) {
-            std::cout << "Group:\n";
-            for (const Beam& beam : group) {
-                std::cout << beam.score << ": " << detokenize(detokenizer, beam.tokens) << '\n';
-            }
-        }
-    }
-    // Model is stateful which means that context (kv-cache) which belongs to a particular
-    // text sequence is accumulated inside the model during the generation loop above.
-    // This context should be reset before processing the next text sequence.
-    // While it is not required to reset context in this sample as only one batch of sequences is processed,
-    // it is called for education purposes:
-    lm.reset_state();
+    auto prompts = std::vector<std::string>(argv + 2, argv + argc);
+    
+    std::string model_path = argv[1];
+    std::string device = "CPU";  // GPU can be used as well
+
+    ov::genai::LLMPipeline pipe(model_path, device);
+    ov::genai::GenerationConfig config = pipe.get_generation_config();
+    config.max_new_tokens = 20;
+    config.num_beam_groups = 3;
+    config.num_beams = 15;
+    config.num_return_sequences = config.num_beams * prompts.size();
+    
+    // for TinyLLama despite in generation_config.json pad_token_id is set to 0, 
```

Comment:
how do we know that?

Reply:
That's a great catch! Here, in `generation_config.json` `pad_token_id` is set to 0 
https://huggingface.co/TinyLlama/TinyLlama-1.1B-Chat-v1.0/blob/main/generation_config.json

but when i tokenize in HF it fills paddings with `</s>` token_id which is 2, as it's defined in `tokenizer_config.json`
https://huggingface.co/TinyLlama/TinyLlama-1.1B-Chat-v1.0/blob/main/tokenizer_config.json#L34

Reply:
agreed to use pad_token_id from tokenizer instead of generation config